std/uni.d

```diff
@@ -7787,6 +7787,41 @@ if (isInputRange!S1 && isSomeChar!(ElementEncodingType!S1)
     import std.typecons : Yes;
     static import std.ascii;
 
+    static if ((isDynamicArray!S1 || isRandomAccessRange!S1)
+        && (isDynamicArray!S2 || isRandomAccessRange!S2)
+        && !(isInfinite!S1 && isInfinite!S2)
+        && __traits(compiles,
+            {
+                size_t s = size_t.sizeof / 2;
+                r1 = r1[s .. $];
+                r2 = r2[s .. $];
```

Comment:
How about `allSatisfy!(hasSlicing, S1, S2)` ?

Reply:
I think for infinite ranges slicing doesn't necessarily produce the same type so I need to check assignment works.